### DISCUSSION TITLE
Vercel REST API : Usage endpoints

### DISCUSSION THREAD

**Discussion by depsimon** [vercel/community#6992] [Feature Question]
### Question

Is it planned to expose usage endpoints in the Vercel REST API?

I'd like to be able to use usage information to do some stuff like showing in a UI or trigger some actions on my own logic.

**Comment by versecafe** [comment]
`https://vercel.com/api/teams?flags=true&permissions=true` will grab all related team data including subscriptions, usage, integrations, etc. There may be some more fine grained api endpoints but this should have everything you need, there is also an RSC that you can parse the data out of at `https://vercel.com/exakty/~/usage` that contains a JSON on the third line which has all your network usage for the top 10 uses.

If I find anything else I'll let you know, but also some improvements to the API for billing & notifications would be great in the future.

**Reply by depsimon** [reply]
Thanks :) The teams endpoint is not complete enough. The usage endpoint is what I'd like but as a REST API. Hopefully this'll become available later

**Reply by leerob** [reply]
What information are you looking for specifically in terms of usage?

**Reply by depsimon** [reply]
I'd like to be able to gauge out KV usage specifically. 

Basically getting a daily report of the KV usage + a slack command.

We're just getting started using Vercel but I can foresee the need for the same kind of reports & commands for other services.

**Reply by versecafe** [reply]
It would be great to be able to get the current and total allowed usage that's behind these charts since it is an RSC serving the data

```json
[
  {
    "name": "string",
    "unit": "string",
    "usage": 0.0,
    "limit": 0.0,
    "cost": 0.0,
    "cost-overshoot": 0.0
  }
]
```

<img width="1394" alt="image" src="https://github.com/vercel/community/assets/147033096/99c51028-5bdd-4bba-b45f-63f80d1ce278">
